src/compiler/checker.ts

```diff
@@ -1852,7 +1852,13 @@ namespace ts {
                         combineValueAndTypeSymbols(symbolFromVariable, symbolFromModule) :
                         symbolFromModule || symbolFromVariable;
                     if (!symbol) {
-                        error(name, Diagnostics.Module_0_has_no_exported_member_1, getFullyQualifiedName(moduleSymbol), declarationNameToString(name));
+                        const suggestion = getSuggestionForNonexistentModule(name, targetSymbol);
+                        if (suggestion !== undefined) {
+                            error(name, Diagnostics.Module_0_has_no_exported_member_1_Did_you_mean_2, getFullyQualifiedName(moduleSymbol), declarationNameToString(name), suggestion);

```

Comment:
Maybe move the first two arguments out of each call to `error` to avoid the duplication in logic.